bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHolmesHandler.java

```diff
@@ -0,0 +1,589 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wemo.internal.handler;
+
+import static org.openhab.binding.wemo.internal.WemoBindingConstants.*;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOParticipant;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOService;
+import org.openhab.binding.wemo.internal.http.WemoHttpCall;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.CharacterData;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * The {@link WemoHolmesHandler} is responsible for handling commands, which are
+ * sent to one of the channels and to update their states.
+ *
+ * @author Hans-Jörg Merk - Initial contribution;
+ */
+
+public class WemoHolmesHandler extends AbstractWemoHandler implements UpnpIOParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(WemoHolmesHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_PURIFIER);
+
+    private Map<String, Boolean> subscriptionState = new HashMap<>();
+
+    private final Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<>());
+
+    private UpnpIOService service;
+
+    private int filterLifeDays = 330;
+    private int filterLifeMins = filterLifeDays * 24 * 60;
+
+    /**
+     * The default refresh interval in Seconds.
+     */
+    private final int DEFAULT_REFRESH_INTERVAL = 120;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private final Runnable refreshRunnable = new Runnable() {
+
+        @Override
+        public void run() {
```

Comment:
Same comment here: java8 notation:
```suggestion
    private final Runnable refreshRunnable = () -> {
```